### PR TITLE
Leaders ladder

### PR DESCRIPTION
**Pull request by krzyurb**
Create leaders ladder for profile plugin.

**Usage example**
User: `profile @user leader`
Bot: `@user's leader is @leader → @leader1 → @leader2 → @leader3`

**JIRA**
TECH-390

### CODE REVIEW

File: src/profile/test/profile.spec.js [hunk 1]
```diff
@@ -163,12 +173,18 @@ describe('Profile test suite', () => {
...
 
-    it('should inform about custom profile field of user type', async () => {
-      await room.user.say('Josh', '@hubot profile @John leader');
-      const leader = room.robot.brain.userForId(user.profile.fields['My leader'].value);
+    it('should respond with leaders ladder', async () => {
+      await room.user.say('Josh', '@hubot profile @John leader')
+      const leader = room.robot.brain.userForId(user.profile.fields.QWERTY.value);
+      const topLeader = room.robot.brain.userForId(leader.profile.fields.QWERTY.value);
+
+      sinon.stub(client.users.profile, 'get').callsFake((params) => {
```
Comment (Getriax) [inline src/profile/test/profile.spec.js]: This array function does not require a body, also `client.users.profile.get` is an async function so please leave it as async one 🙂

File: src/profile/src/profile.service.js [hunk 2]
```diff
@@ -54,10 +59,35 @@ class ProfileService {
...
       return true;
     }
+
     const customFields = await this.getCustomFields();
     return customFields.map(field => field.label).includes(askedField);
   }
 
+  async leadersLadder(id) {
+    const { profile } = await this.client.users.profile.get({ user: id });
+    const leaderField = process.env.LEADER_FIELD_ID;
```
Comment (Getriax) [inline src/profile/src/profile.service.js]: Please add this environment variable to the documentation 🙂

File: src/profile/test/profile.spec.js [hunk 3]
```diff
@@ -163,12 +173,18 @@ describe('Profile test suite', () => {
...
+        return room.robot.brain.userForId(params.user);
+      });
+
       const expected = [
         ['Josh', '@hubot profile @John leader'],
-        ['hubot', `@John's leader is @${leader.name}`],
+        ['hubot', `@John's leader is @${leader.name} → @${topLeader.name}`],
```
Comment (Getriax) [inline src/profile/test/profile.spec.js]: What do you think about changing the `@John's leader is` to `@John's leaders are`? There are multiple people mentioned 🙂

**Comment by krzyurb** [comment]
@Getriax fixes done

**Pull request merged**